pkg/server/cloud/aws/network.go

```diff
@@ -316,57 +316,85 @@ func (e *AwsEC2) ModifySourceDestinationCheck(instanceID string, isEnabled bool)
 	return err
 }
 
-func (e *AwsEC2) RemoveRoute(destinationCIDR string) error {
+func (e *AwsEC2) RemoveRoute(destinationCIDR, instanceID string) error {
+	if destinationCIDR == "" || instanceID == "" {
+		return fmt.Errorf(
+			"invalid input: at least one non-empty value needed (got %q %q)",
+			destinationCIDR, instanceID)
+	}
 	out, err := e.client.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
 		Filters: []*ec2.Filter{
 			{
-				Name: aws.String("vpc-id"),
-				Values: []*string{
-					aws.String(e.vpcID),
-				},
+				Name:   aws.String("vpc-id"),
+				Values: aws.StringSlice([]string{e.vpcID}),
+			},
+			{
+				Name:   aws.String("association.main"),
+				Values: aws.StringSlice([]string{"false"}),
+			},
+			{
+				Name:   aws.String("association.subnet-id"),
+				Values: aws.StringSlice([]string{e.subnetID}),
 			},
 		},
 	})
 	if err != nil {
-		return util.WrapError(err, "Could not list vpc route tables")
+		return util.WrapError(err, "listing vpc route tables")
 	}
 	for _, table := range out.RouteTables {
 		if len(table.Associations) == 0 {
 			continue
 		}
 		for _, route := range table.Routes {
-			if destinationCIDR == aws.StringValue(route.DestinationCidrBlock) {
-				_, err = e.client.DeleteRoute(&ec2.DeleteRouteInput{
-					DestinationCidrBlock: route.DestinationCidrBlock,
-					RouteTableId:         table.RouteTableId,
-				})
-				if err != nil {
-					return util.WrapError(
-						err,
-						"Error deleting old route table entry for table %s",
-						*table.RouteTableId,
-					)
-				}
-				break
+			if destinationCIDR != "" &&
```

Comment:
RemoveRoute() now can handle 4 scenarios:
- destinationCIDR and instanceID are both set, it will remove routes that point to the instance for the prefix in destination CIDR.
- destinationCIDR empty, instanceID set: it will remove routes (any prefix) that point to the instance.
- destinationCIDR set, instanceID empty: it will remove routes for the prefix in destinationCIDR, if it finds an instance route (for any instance).
- destinationCIDR and instanceID are both empty: it will remove blackhole instance routes (i.e. any inactive route pointing to an instance).

So destinationCIDR can be empty here.